util/HalideTraceUtils.h

```diff
@@ -0,0 +1,91 @@
+#ifndef HALIDE_TRACE_UTILS_H
+#define HALIDE_TRACE_UTILS_H
+
+#include "HalideRuntime.h"
+#include <stdio.h>
+
+namespace Halide {
+namespace Internal {
+
+void bad_type_error(halide_type_t type);
+
+// Simple conversion.
+template<typename T>
+T value_as(halide_type_t type, const void* value) {
+    switch (type.code) {
+    case halide_type_int:
+        switch (type.bits) {
+        case 8:
+            return (T)(((const int8_t *)value)[0]);
+        case 16:
+            return (T)(((const int16_t *)value)[0]);
+        case 32:
+            return (T)(((const int32_t *)value)[0]);
+        case 64:
+            return (T)(((const int64_t *)value)[0]);
+        default:
+            bad_type_error(type);
+        }
+        break;
+    case halide_type_uint:
+        switch (type.bits) {
+        case 8:
```

Comment:
are you deliberately ignoring uint(1) (aka bool)?

Reply:
This is just code from Sander's PR. Probably an oversight - will add it.

Reply:
(Actually looks like it was my code originally)